### PR TITLE
chore: refine the logic of source manager

### PR DESCRIPTION
**Pull request by shanicky**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

This PR refactors the source manager

1. remove the meta store in the source manager and store actor splits in the actor_splits field of the fragment in the fragment manager, so that they can be stored in one transaction during the scaling process and other senario.s

2. introduce the Command of SourceSplitAssignment

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

### CODE REVIEW

File: src/meta/src/manager/catalog/fragment.rs [hunk 1]
```diff
@@ -466,6 +467,33 @@ where
...
+        let map = &mut self.core.write().await.table_fragments;
+        let mut transaction = Transaction::default();
+
+        for (_, table_fragments) in map.iter_mut() {
+            let mut updated = false;
+
+            for fragment_id in table_fragments.fragments.keys() {
+                if let Some(actor_splits) = split_assignment.get(fragment_id).cloned() {
+                    updated = true;
+                    table_fragments.actor_splits.extend(actor_splits);
```
[the remark this answers is not in the document]
Comment (shanicky) [reply src/meta/src/manager/catalog/fragment.rs:483]: Here `split_assignment` is indeed the full splits, but only within this fragment, maybe the source of another fragment in the same table has not changed. There should be a problem here, we can't extend directly, maybe some actor was deleted, there should be an assert

[the remark this answers is not in the document]
Comment (yezizp2012) [reply src/meta/src/manager/catalog/fragment.rs:483]: If `update_actor_splits_by_split_assignment` will only be called when config change, I think that won't be a problem, the actors won't get deleted and extend is okay here. I agree please add an assertion here.

**Review by yezizp2012 — APPROVED** [review]
LGTM

**Review by tabVersion — APPROVED** [review]
LGTM

**Pull request merged**